asapdiscovery-data/asapdiscovery/data/openeye.py

```diff
@@ -126,6 +127,30 @@ def load_openeye_pdb(pdb_fn, alt_loc=False):
         oechem.OEThrow.Fatal(f"Unable to open {pdb_fn}")
 
 
+def load_openeye_cif1(fn: str) -> oechem.OEGraphMol:
+    """
+    Loads a biological assembly file into an OEGraphMol object.
+    Current version requires going through an OpenMM intermediate.
+
+    Args:
+    - fn (str): the filename of the biological assembly file.
+
+    Returns:
+    - oechem.OEGraphMol: the biological assembly as an OEGraphMol object.
+    """
+    from tempfile import NamedTemporaryFile
+
+    from openmm.app import PDBFile, PDBxFile
+
+    cif = PDBxFile(fn)
+
+    # the keep ids flag is critical to make sure the residue numbers are correct
+    with NamedTemporaryFile("w", suffix=".pdb") as f:
```

Comment:
I made an OpenEye ticket for this:
>I wanted to come back to you with a quick update on this ticket. After some investigation, the developer realized that you are using the biological assembly (even though you clearly stated that) from the RCSB and not the asymmetric unit PDB file. Simply put, Spruce is not meant to be used with assembly PDBs. Spruce generates the proper assemblies as specified in the asymmetric unit PDBs in the same way that the RCSB does, except we have more options and features for this processes. We would encourage you to retry with the asymmetric unit PDB file and please let us know if you are still encountering any issues. Lastly, you may have a specific reason to be using the biological assembly files, which is totally fair! If this is the case, we would be interested to know why so that we could take this into consideration for future Spruce improvements.

Reply:
Basically I never figured out how to properly make the dimer from the monomer PDBs using OpenEye, and you can download the Biological Assemblies from the PDB as CIFs, and OpenMM manages to handle this reasonably well, as far as I can tell.